### PR TITLE
Adding support for support assist

### PR DESCRIPTION
**Pull request by trisha-dell**
# Description
Adding support for Support Assist feature of PowerScale 

# Checklist:

- [X] I have performed a self-review of my code to ensure there are no formatting, pep8, linting, or security issues
- [X] I have performed the Ansible Sanity test using --docker default
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Functional Testing
- [X] Unit testing

### CODE REVIEW

File: plugins/modules/support_assist.py [hunk 1]
```diff
@@ -0,0 +1,772 @@
...
+                            last_name=dict(type='str'),
+                            phone=dict(type='str'))
+                    ),
+                    secondary=dict(
+                        type='dict', options=dict(
+                            email=dict(type='str'),
+                            first_name=dict(type='str'),
+                            last_name=dict(type='str'),
+                            phone=dict(type='str'))
+
```
Comment (Jennifer-John) [inline plugins/modules/support_assist.py:701]:
```suggestion
```

File: playbooks/modules/support_assist.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    - name: Accept support assist terms
+      dellemc.powerscale.support_assist:
+        onefs_host: "{{ onefs_host }}"
+        port_no: "{{ port_no }}"
+        api_user: "{{ api_user }}"
+        api_password: "{{ api_password }}"
+        verify_ssl: "{{ verify_ssl }}"
+        accepted_terms: true
+
+    - name: Get support assist settings
```
[the remark this answers is not in the document]
Comment (trisha-dell) [reply playbooks/modules/support_assist.yml:22]: Yes, the module also supports it.

File: plugins/modules/info.py [hunk 3]
```diff
@@ -2153,6 +2164,184 @@
...
+                    type: str
+                network_pools:
+                    description: List of network pools.
+                    type: list
+                    elements: dict
+                    contains:
+                        pool_name:
+                            description: Name of the network pool.
+                            type: str
+                        state:
```
[the remark this answers is not in the document]
Comment (trisha-dell) [reply plugins/modules/info.py:2216]: state is not part of the return values as this is a module param.

File: tests/unit/plugins/module_utils/mock_info_api.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright: (c) 2021, Dell Technologies
+# Copyright: (c) 2021=2024, Dell Technologies
```
Comment (Jennifer-John) [inline tests/unit/plugins/module_utils/mock_info_api.py:1]:
```suggestion
# Copyright: (c) 2021-2024, Dell Technologies
```

File: plugins/modules/support_assist.py [hunk 5]
```diff
@@ -0,0 +1,771 @@
...
+    def get_support_assist_parameters(self):
+        return dict(
+            accepted_terms=dict(type='bool'),
+            automatic_case_creation=dict(type='bool'),
+            connection=dict(
+                type='dict', options=dict(
+                    gateway_endpoints=dict(
+                        type='list', elements='dict', options=dict(
+                            enabled=dict(type='bool', default=True),
+                            gateway_host=dict(type='str'),
```
Comment (Jennifer-John) [inline plugins/modules/support_assist.py:669]: can we make gateway_host in gateway_endpoints as a required param

File: tests/unit/plugins/modules/test_support_assist.py [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+from ansible_collections.dellemc.powerscale.plugins.modules.support_assist import SupportAssist
+from ansible_collections.dellemc.powerscale.plugins.modules.support_assist import SupportAssistHandler
+from ansible_collections.dellemc.powerscale.plugins.modules.support_assist import main
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.mock_support_assist_api \
+    import MockSupportAssistApi
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.mock_api_exception \
+    import MockApiException
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.shared_library.powerscale_unit_base \
+    import PowerScaleUnitBase
+from ansible_collections.dellemc.powerscale.tests.unit.plugins.module_utils.mock_sdk_response \
```
Comment (Bhavneet-Sharma) [inline tests/unit/plugins/modules/test_support_assist.py:25]: If not used please do remove

File: tests/unit/plugins/modules/test_support_assist.py [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+            {
+                "connection": {
+                    "network_pools": [
+                        {
+                            "pool_name": "subnet0:pool1",
+                            "state": "absent"
+                        },
+                        {
+                            "pool_name": "subnet0:pool5",
+                            "state": "present"
```
Comment (Bhavneet-Sharma) [inline tests/unit/plugins/modules/test_support_assist.py:64]: Use variable if it repeated more than thrice to eliminate the sonar issues.

File: plugins/modules/support_assist.py (hunk 5 above)
Comment (trisha-dell) [reply plugins/modules/support_assist.py]: done.

File: tests/unit/plugins/modules/test_support_assist.py (hunk 7 above)
Comment (trisha-dell) [reply tests/unit/plugins/modules/test_support_assist.py:64]: used just once.

File: plugins/modules/info.py [hunk 8]
```diff
@@ -2889,7 +3082,8 @@ def perform_module_operation(self):
             ClusterIdentity=cluster_identity,
             ClusterOwner=cluster_owner,
             SnmpSettings=snmp_settings,
-            ServerCertificate=server_certificate
+            ServerCertificate=server_certificate,
+            SupportAssistSettings=support_assist_settings
```
Comment (sachin-apa) [inline plugins/modules/info.py:3086]: we had a comments from anupam on the naming conventions, should we name this as support_assist_settings instead ?

**Review by sachin-apa — APPROVED** [review]

**Review by felixs88 — APPROVED** [review]

**Pull request merged**